lib/rexml/parsers/baseparser.rb

```diff
@@ -471,9 +471,11 @@ def pull_event
                 end
 
                 return [ :comment, md[1] ]
-              else
-                md = @source.match(/\[CDATA\[(.*?)\]\]>/um, true)
-                return [ :cdata, md[1] ] if md
+              elsif @source.match?("[CDATA[", true)
+                text = @source.read_until("]]>")
+                if text.chomp!("]]>")
+                  return [ :cdata, text ]
```

Comment:
Can we add a `else` clause for not-closed `<![CDATA[`?

Reply:
OK.
I see.